bindsnet/network/network.py

```diff
@@ -100,7 +100,8 @@ def __init__(
         self.layers = {}
         self.connections = {}
         self.monitors = {}
-        self.learning = learning
+        self.train(learning)
```

Comment:
What's your reason for changing this?

Reply:
For backwards compatibility `self.learning` is kept throughout the code and `self.train()` (the method from `torch.nn.Module`) has been overloaded to include the writing of `self.learning` as these effectively do the same operation. We can use `network.train()` or `network.eval()` as pytorch uses them, however directly writing to `network.training` no longer works. Pytorch internally keeps track of `self.training`. We could switch completely without too much work.

Reply:
Yeah, that's a good point.

Reply:
What Pytorch internally keeps track of when using `self.training`?
How that it help us? beside been more compatible with PyTorch API?

Reply:
https://github.com/pytorch/pytorch/blob/181792176d46b5e14b4d4c63292afb6670f5e27e/torch/nn/modules/module.py#L987

it is used for batchnorm and dropout mainly. When switching between training and testing people are used to calling `train()` and `eval()` on their models. It is just a function that sets the flag in every `torch.nn.Module` that is a child of the primary one that it was called from.